### ISSUE TITLE
Make Mie and Polynomials compatible with XP

### ISSUE THREAD

**Issue by giovannivolpe** [DeepTrackAI/DeepTrack2#339]
Make the modules `mie.py` and `polynomials.py` compatible with XP.

Note: scipy.special does not maintain gradients, so we need to implement the three functions ourselves. But that should not be difficult.

**Comment by Pwhsky** [comment]
## Potential Solutions

[`torch.special`](https://docs.pytorch.org/docs/stable/special.html) would be ideal but it only supports Bessel functions of order `0` and `1` and lacks Riccati-Bessel functions. But would otherwise be perfect as it has full Torch integration.

Another possible solution is combining `torch.autograd.Function` with `scipy.special` and encapsulating the polynomial functions in a wrapper class, something like:

```python 

class BesselWrapper(torch.autograd.Function):
    def forward(l, x):
        "Call scipy.special here"
        y = scipy.special.jv(l, x)
        return torch.tensor(y)

    def backward(context, grad):
        return "backward function"

def besselj(l, x):
    if isinstance(x, np.ndarray): # if numpy
        return scipy.special.jv(l, x)
    else: #if torch
        return BesselWrapper.apply(l, x)
```
Or if there are more elegant branchless solutions we should use those.

**Comment by BenjaminMidtvedt** [comment]
My main concerns are that
1. We may lose the ability to run it on the gpu
2. We dtill have to implement the backwards function. Is it then not easier to just implement the forward function? 

We only need three classes of bessel/hankel polynomials, not the entire scipy.special. the rest are already implemented in mie.py derived from these three
